src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.tsx

```diff
@@ -21,45 +21,21 @@ export interface TooltipOverlayProps {
   onClose(): void;
 }
 
-type CombinedProps = TooltipOverlayProps & {
-  i18n: ReturnType<typeof useI18n>;
-};
-
-class TooltipOverlayInner extends PureComponent<CombinedProps, never> {
```

Comment:
I removed the pure component aspect since it's unlikely the `activator` prop will be memorized and we're *probably* getting very little gain from it.

Reply:
[Death to pure components!](https://github.com/Shopify/polaris-react/issues/2062)